File: Android/session/src/main/java/com/felipearpa/tyche/session/AuthTokenFirebaseRetriever.kt
package com.felipearpa.tyche.session

import com.google.firebase.auth.FirebaseAuth
import kotlinx.coroutines.tasks.await

class AuthTokenFirebaseRetriever(private val firebaseAuth: FirebaseAuth) :
    AuthTokenRetriever {
    override suspend fun authToken(): String? {
        return firebaseAuth.currentUser?.getIdToken(true)?.await()?.token
    }
}

File: iOS/Session/Sources/Session/AuthTokenFirebaseRetriever.swift
@preconcurrency import FirebaseAuth

final class AuthTokenFirebaseRetriever: AuthTokenRetriever {
    private let firebaseAuth: Auth
    
    init(firebaseAuth: Auth) {
        self.firebaseAuth = firebaseAuth
    }
    
    func authToken() async -> String? {
        try? await firebaseAuth.currentUser?.getIDToken()
    }
}
